packages/safe-ds-vscode/src/extension/eda/apis/runnerApi.ts

```diff
@@ -0,0 +1,415 @@
+    }
+
+    private sdsStringForIDnessByColumnName(columnName: string, tablePlaceholder: string, newPlaceholderName: string) {
+        return (
+            'val ' + newPlaceholderName + ' = ' + tablePlaceholder + '.get_column("' + columnName + '").idness(); \n'
+        );
+    }
+
+    private sdsStringForHistogramByColumnName(
+        columnName: string,
+        tablePlaceholder: string,
+        newPlaceholderName: string,
+    ) {
+        return (
+            'val ' +
+            newPlaceholderName +
+            ' = ' +
+            tablePlaceholder +
+            '.get_column("' +
+            columnName +
+            '").plot_histogram(); \n'
+        );
+    }
+
+    // --- Placeholder handling ---
+
+    private genPlaceholderName(): string {
+        return CODEGEN_PREFIX + this.placeholderCounter++;
+    }
+
+    private async getPlaceholderValue(placeholder: string, pipelineExecutionId: string): Promise<any | undefined> {
+        return new Promise((resolve) => {
+            if (placeholder === '') {
+                resolve(undefined);
+            }
+
+            const placeholderValueCallback = (message: messages.PlaceholderValueMessage) => {
+                if (message.id !== pipelineExecutionId || message.data.name !== placeholder) {
+                    return;
+                }
+                this.services.runtime.Runner.removeMessageCallback(placeholderValueCallback, 'placeholder_value');
+                resolve(message.data.value);
+            };
+
+            this.services.runtime.Runner.addMessageCallback(placeholderValueCallback, 'placeholder_value');
+            printOutputMessage('Getting placeholder from Runner ...');
+            this.services.runtime.Runner.sendMessageToPythonServer(
+                messages.createPlaceholderQueryMessage(pipelineExecutionId, placeholder),
+            );
+
+            setTimeout(() => {
+                resolve(undefined);
+            }, 30000);
+        });
+    }
+
+    // --- Public API ---
+
+    public async getTableByPlaceholder(tableName: string, pipelineExecutionId: string): Promise<Table | undefined> {
+        const pythonTableColumns = await this.getPlaceholderValue(tableName, pipelineExecutionId);
+        if (pythonTableColumns) {
+            const table: Table = {
+                totalRows: 0,
+                name: tableName,
+                columns: [] as Table['columns'],
+                appliedFilters: [] as Table['appliedFilters'],
+            };
+
+            let i = 0;
+            let currentMax = 0;
+            for (const [columnName, columnValues] of Object.entries(pythonTableColumns)) {
+                if (!Array.isArray(columnValues)) {
+                    continue;
+                }
+                if (currentMax < columnValues.length) {
+                    currentMax = columnValues.length;
+                }
+
+                const isNumerical = typeof columnValues[0] === 'number';
```

Comment:
Even if the first entry of a column is numeric, the entire column might still be categorical. For example, the column might contain values 0 for false and 1 for true.

Reply:
How to get this type then. Can it also be passed with the table somehow from Runner?

Reply:
So it would need extra calls, I see. What would I use the idness for in this context? To display a column with just 0 and 1 as categorical you mean? I would personally just use the types the columns actually have in the background and not interpret that myself